### DISCUSSION TITLE
[mixed time-series] Separate Y-axis bounds, Label values and Tooltip formatting

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18494] [Ideas]
**Is your feature request related to a problem? Please describe.**
1. Currently there is an option to set Y-axis bounds, but it applies to both primary and secondary y-axis at the same time. This is not really helpful since both axis will mostly have different scales.
2. We cannot see the value labels in the chart, so the user always has to rely on tooltip to look at the data.
3. Tooltip is always showing adaptive formatting for X-axis and Secondary Y-axis values.

**Describe the solution you'd like**
1. There should be an option to set bounds separately for both primary and secondary y-axis.
2. There should be an option to show value labels(currently available only in Bar chart)
3. d3 formatting set by user should be shown in tooltip for all the axis values.

**Describe alternatives you've considered**

**Additional context**

**Comment by junlincc** [comment]
thanks for suggesting! all good point and well accepted. Let us know if would like to collaborate on this, as it will take awhile for us to start the next iteration that incorporates those changes. 🙏

**Comment by kamalkeshavani-aiinside** [comment]
@junlincc Thanks for your comment. Although I don't have any experience with type script, I am hoping to get some similar fixes from other charts and re-use the code for this chart.
I am starting by looking at https://github.com/apache/superset/blob/master/CONTRIBUTING.md#improving-visualizations. I guess I can ask questions in #contributing channel in slack whenever I get stuck somewhere.

**Comment by serenajiang** [comment]
One of our users reported a similar problem - I wonder whether this issue should be considered a bug fix rather than an enhancement request? The tooltip always uses the primary y-axis formatting even if there is a different format specified for the secondary y-axis. This is a problem for charts with very different scales (ex. proportions vs large integers).

**Comment by kevin-struckhoff** [comment]
Additionally, for single Y-axis charts, it would be helpful to allow the user to set custom grid line interval values. The attached screenshot from OpenOffice Spreadsheet shows how they handle it.
![Y-axis-interval-setting](https://user-images.githubusercontent.com/88514509/129094670-5f439b4e-d488-4798-ad95-b98ea6d2b5d5.png)

**Comment by villebro** [comment]
> The tooltip always uses the primary y-axis formatting even if there is a different format specified for the secondary y-axis. This is a problem for charts with very different scales (ex. proportions vs large integers).

FYI @serenajiang this problem should have been fixed by https://github.com/apache-superset/superset-ui/pull/1143

**Comment by kamalkeshavani-aiinside** [comment]
> 1. There should be an option to set bounds separately for both primary and secondary y-axis.
> 2. There should be an option to show value labels(currently available only in Bar chart)
> 3. d3 formatting set by user should be shown in tooltip for all the axis values.

2 & 3 are now fixed in master. Waiting for 1st.
And for 2nd, Y-axis values can be shown in chart, but they don't follow the formatting set by the user.
![image](https://user-images.githubusercontent.com/74634977/143389096-63f39e88-c8e3-496a-a671-33f6c5616e47.png)

**Comment by drummerwolli** [comment]
to close the loop.

> 1. There should be an option to set bounds separately for both primary and secondary y-axis.

This was added in: https://github.com/apache/superset/pull/23917

and it's released since 3.0.0: https://github.com/apache/superset/blob/master/CHANGELOG/3.0.0.md
